sklearn/tests/test_base.py

```diff
@@ -143,6 +145,24 @@ def test_clone_nan():
     assert_true(clf.empty is clf2.empty)
 
 
+def test_clone_sparse_matrices():
+    sparse_matrix_classes = [
+        getattr(sp, name)
+        for name in dir(sp) if name.endswith('_matrix')]
+
+    PY26 = sys.version_info[:2] == (2, 6)
+    if PY26:
+        # sp.dok_matrix can not be deepcopied in Python 2.6
```

Comment:
For the record:

```py
from copy import deepcopy

import numpy as np

from scipy import sparse as sp

m = sp.dok_matrix(np.eye(5))

deepcopy(m)
```

```
AttributeError: shape not found
```

For some reason it looks like the reconstructed matrix is missing some attributes. I think it's fine not testing dok_matrix for Python 2.6.